whylog/constraints/verifier.py

```diff
@@ -144,37 +156,70 @@ def constraints_or(cls, clues_lists, effect, constraints, constraint_manager):
         and for each such combination produces InvestigationResult object.
         returns list of all produced InvestigationResults
         """
-        clues_lists = cls._construct_proper_clues_lists(clues_lists)
         causes = []
-        for combination in cls._clues_combinations(clues_lists):
-            verified_constraints = [
-                constraint
-                for constraint in constraints
-                if cls._verify_constraint(combination, effect, constraint, constraint_manager)
-            ]
-            if len(verified_constraints) > 0:
-                causes.append(
-                    cls._create_investigation_result(
-                        [
-                            clue for clue in combination if not clue == Verifier.UNMATCHED
-                        ], verified_constraints
+        if not constraints:
+            # when there is lack of constraints, but there are existing clues combinations,
+            # each of them should be returned
+            for combination in cls._clues_combinations(clues_lists):
```

Comment:
```python
return [cls._pack_results_for_constraint_or(combination, constraints) for combination in cls._clues_combinations(clues_lists)]
```
to increase performance.
Then the rest of the function doesn't need to be in the `else` block.